odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataSourceServiceImpl.java

```diff
@@ -11,60 +12,135 @@
 import org.opendatadiscovery.oddplatform.dto.TokenDto;
 import org.opendatadiscovery.oddplatform.exception.NotFoundException;
 import org.opendatadiscovery.oddplatform.mapper.DataSourceMapper;
-import org.opendatadiscovery.oddplatform.repository.DataSourceRepository;
+import org.opendatadiscovery.oddplatform.model.tables.pojos.DataSourcePojo;
+import org.opendatadiscovery.oddplatform.model.tables.pojos.NamespacePojo;
 import org.opendatadiscovery.oddplatform.repository.TokenRepository;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveDataEntityRepository;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveDataSourceRepository;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveNamespaceRepository;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveSearchEntrypointRepository;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 @Service
+@RequiredArgsConstructor
 @Slf4j
-public class DataSourceServiceImpl
-    extends
-    AbstractCRUDService<DataSource, DataSourceList, DataSourceFormData, DataSourceUpdateFormData,
-        DataSourceDto, DataSourceMapper, DataSourceRepository>
-    implements DataSourceService {
+public class DataSourceServiceImpl implements DataSourceService {
```

Comment:
Mono.zip(
                            searchEntrypointRepository.updateDataSourceVector(dsDto.dataSource().getId()),
                            searchEntrypointRepository.updateNamespaceVector(dsDto.namespace().getId())
                        )

can be extracted